presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -371,7 +371,14 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                 ImmutableList.Builder<JoinNode.EquiJoinClause> joinConditionBuilder = ImmutableList.builder();
                 ImmutableList.Builder<Expression> joinFilterBuilder = ImmutableList.builder();
                 for (Expression conjunct : extractConjuncts(newJoinPredicate)) {
-                    if (joinEqualityExpression(node.getLeft().getOutputSymbols()).test(conjunct)) {
+                    ComparisonExpressionType equiComparison = null;
```

Comment:
instead of doing this, why don't you just change the joinEqualityExpression to check for IS_NOT_DISTINCT_FROM, and extract the type from the cast ComparisonExpression?